components/base/sensorcontrolled/sensorcontrolled_test.go

```diff
@@ -258,17 +258,17 @@ func TestReconfig(t *testing.T) {
 	test.That(t, err, test.ShouldBeNil)
 	sb, ok := b.(*sensorBase)
 	test.That(t, ok, test.ShouldBeTrue)
-	test.That(t, sb.orientation.Name().ShortName(), test.ShouldResemble, "orientation")
+	headingOri, headingSupported, err := sb.headingFunc(context.Background())
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, headingSupported, test.ShouldBeTrue)
+	test.That(t, headingOri, test.ShouldEqual, orientationValue)
 
 	deps, cfg = msDependencies(t, []string{"orientation1"})
 	err = b.Reconfigure(ctx, deps, cfg)
 	test.That(t, err, test.ShouldBeNil)
-	test.That(t, sb.orientation.Name().ShortName(), test.ShouldResemble, "orientation1")
-
-	deps, cfg = msDependencies(t, []string{"orientation2"})
-	err = b.Reconfigure(ctx, deps, cfg)
 	test.That(t, err, test.ShouldBeNil)
-	test.That(t, sb.orientation.Name().ShortName(), test.ShouldResemble, "orientation2")
```

Comment:
@randhid should I record the heading sensors name somewhere or can I gut alot of these tests too